test/puppetlabs/puppetdb/threadpool_test.clj

```diff
@@ -32,14 +32,39 @@
 
       (with-open [gtp (gated-threadpool 1 "test-pool-%d" 1000)]
         (let [fut (future
-                    (dochan gtp worker-fn in-chan))]
+                    (dochan gtp worker-fn in-chan tu/blocked?))]
           (async/>!! in-chan {:id :a})
           (is (= [:done :a] (async/<!! out-chan)))
           (is (= 1 @counter))
           (async/close! in-chan)
           (is (not= ::timed-out (deref fut tu/default-timeout-ms ::timed-out)))
           (is (true? (future-done? fut)))))))
 
+
+  (testing "block command"
+    (let [counter (atom 0)
+          in-chan (async/chan 10)
+          out-chan (async/chan 10)
+          worker-fn (wrap-out-chan
+                      out-chan
+                      (fn [{:keys [id]}]
+                        (swap! counter inc)
+                        [:done id]))
+          blocked? (atom true)]
+
+        (with-open [gtp (gated-threadpool 1 "test-pool-%d" 1000)]
+          (let [fut (future
```

Comment:
Suggest in most cases using a `noisy-future` so we'll make sure to see any errors, unless we know we're going to always deref it, and we want to be able to see the exception then.

Reply:
Replaced `future` with `noisy-future`